File: packages/brownfield-navigation/android/src/main/java/com/callstack/nativebrownfieldnavigation/BrownfieldNavigationManager.kt
package com.callstack.nativebrownfieldnavigation

object BrownfieldNavigationManager {
  private var navigationDelegate: BrownfieldNavigationDelegate? = null

  fun setDelegate(navigationDelegate: BrownfieldNavigationDelegate) {
    this.navigationDelegate = navigationDelegate
  }

  fun getDelegate(): BrownfieldNavigationDelegate {
    return navigationDelegate
      ?: throw IllegalStateException("BrownfieldNavigation delegate is not set.")
  }
}

File: packages/brownfield-navigation/ios/BrownfieldNavigationManager.swift
//
//  BrownfieldNavigationManager.swift
//
//  Created by Hur Ali on 10/02/2026.
//

public class BrownfieldNavigationManager: NSObject {
  @objc public static let shared = BrownfieldNavigationManager()
  private var navigationDelegate: BrownfieldNavigationDelegate?

  public func setDelegate(navigationDelegate: BrownfieldNavigationDelegate) {
    self.navigationDelegate = navigationDelegate
  }
  
  @objc public func getDelegate() -> BrownfieldNavigationDelegate {
    guard let delegate = navigationDelegate else {
      fatalError("BrownfieldNavigationDelegate not set. Call setDelegate() before using navigation.")
    }
    return delegate
  }
}
